dissect/ntfs/secure.py

```diff
@@ -38,6 +39,9 @@ def __init__(self, record: MftRecord = None, sds: BinaryIO = None):
         if not hasattr(self.sds, "size"):
             self.sds.size = self.sds.seek(0, io.SEEK_END)
 
+        self.lookup = lru_cache(4096)(self.lookup)
+        self._iter_entries = lru_cache(4096)(self._iter_entries)
```

Comment:
```suggestion
```
`_iter_entries` returns a generator so unfortunately caching will not work as you might think here. Plus `lookup` is the primary method anyway, and what I think has the most benefit of caching.

Reply:
Removed it